pyproject.toml

```diff
@@ -158,3 +148,46 @@ platforms = ["linux-64", "osx-64", "osx-arm64", "win-64", "osx-arm64", "linux-aa
 # This is necessary to pull in the lockfile/filelock dependency
 # since we don't handle the optional dependency.
 cachecontrol-with-filecache = ">=0.12.9"
+
+
+[tool.ruff]
+target-version = "py38"
+ignore = [
+    "E501",
+    "F401",
+    "F403",
+    # Disabled during migration to Ruff:
+    "A001",
+    "A002",
+    "A003",
+    "C401",
+    "C405",
+    "C408",
+    "C409",
+    "C413",
+    "C414",
+    "C416",
+    "RUF012",
+    "RUF015",
+]
+line-length = 89
+select = [
+    "A",  # flake8-builtins
+    # "B",  # flake8-bugbear
+    "B006",  # Do not use mutable data structures for argument defaults
```

Comment:
`flake8-mutable` errors are caught by this rule. Looks to have detected something that was previously undetected, too.